bigbluebutton-html5/imports/ui/components/external-video-player/component.jsx

```diff
@@ -39,8 +44,23 @@ class VideoPlayer extends Component {
     window.addEventListener('resize', this.resizeListener);
   }
 
-  componentDidUpdate(nextProps) {
-    if (!nextProps.videoId) {
+  componentDidUpdate(prevProps) {
+    const { inEchoTest } = this.props;
+    const {
+      mutedByEchoTest,
+    } = this.state;
+
+    if (inEchoTest && !this.player.isMuted() && !mutedByEchoTest) {
+      this.player.mute();
+      this.changeState(true);
```

Comment:
Isn't it better to also pass the state property you're modifying? I know that we only have 1 here but still...
Same for setting it to false on line 60

Reply:
Never mind - I see `changeState` is a method you defined in the same file. Is it necessary? It tricked me, maybe others too

Reply:
I did that because linter don't allow declare `setState` inside a `componentDIdUpdate`

Reply:
Sounds like you were able to trick it but is there a real alternative to initiating a state update inside `componentDidUpdate`?